### PR TITLE
fix: use cpuCount for all parallel parts

### PR DESCRIPTION
**Pull request by wardpeet**
## Description

Not every part of Gatsby is using `cpuCoreCount` and we should always do -1 to make sure the main process is left alone.

### CODE REVIEW

**Comment by axe312ger** [comment]
More cores used on minify? I like!

File: packages/gatsby-plugin-manifest/src/gatsby-node.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ sharp.simd(true)
 // Handle Sharp's concurrency based on the Gatsby CPU count
 // See: http://sharp.pixelplumbing.com/en/stable/api-utility/#concurrency
 // See: https://www.gatsbyjs.org/docs/multi-core-builds/
-sharp.concurrency(cpuCoreCount())
+sharp.concurrency(cpuCoreCount() - 1)
```
Comment (vladar) [inline packages/gatsby-plugin-manifest/src/gatsby-node.js:15]: Does concurrency make sense for the manifest plugin at all? It is for a single icon resizing, right?

Comment (KyleAMathews) [reply packages/gatsby-plugin-manifest/src/gatsby-node.js:15]: yeah — it wouldn't ever use more than one core

Comment (wardpeet) [reply packages/gatsby-plugin-manifest/src/gatsby-node.js:15]: Problem here is that sharp is a singleton so it will be shared with gatsby-plugin-sharp :'(

Comment (vladar) [reply packages/gatsby-plugin-manifest/src/gatsby-node.js:15]: But `gatsby-plugin-sharp` now has concurrency 1 (after this PR: https://github.com/gatsbyjs/gatsby/pull/28575):

https://github.com/gatsbyjs/gatsby/blob/f628c3801226f5b500d74b5b41bca4caa72a9e00/packages/gatsby-plugin-sharp/src/process-file.js#L19-L20

Comment (vladar) [reply packages/gatsby-plugin-manifest/src/gatsby-node.js:15]: I wonder what is the actual behavior currently if you have both plugins 🤔

Comment (wardpeet) [reply packages/gatsby-plugin-manifest/src/gatsby-node.js:15]: oh interesting :D I'll set it to one than. Basically last one in gatsby-config wins

**Comment by vladar** [comment]
`cpuCoreCount() - 1` can be equal to `0` with one core and also when `GATSBY_CPU_COUNT` env variable is set to `1`

**Review by vladar — APPROVED** [review]
👍

**Pull request merged**